### PR TITLE
ECL-600: Liability date in PDF

### PR DESCRIPTION
**Pull request by paddynski-opencast**

### CODE REVIEW

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/CheckYourAnswersController.scala [hunk 1]
```diff
@@ -62,7 +62,9 @@ class CheckYourAnswersController @Inject() (
     ).flatten
   ).withCssClass("govuk-!-margin-bottom-9")
 
-  private def organisationDetails()(implicit request: RegistrationDataRequest[_]): SummaryList =
+  private def organisationDetails(liabilityRow: Option[SummaryListRow] = None)(implicit
```
Comment (liandra-giacomine) [inline app/uk/gov/hmrc/economiccrimelevyregistration/controllers/CheckYourAnswersController.scala:65]: why are you passing it in as a parameter as opposed to just adding the row in the Seq?

File: test/uk/gov/hmrc/economiccrimelevyregistration/controllers/CheckYourAnswersControllerSpec.scala [hunk 2]
```diff
@@ -443,4 +448,35 @@ class CheckYourAnswersControllerSpec extends SpecBase {
         }
     }
   }
+
+  "createAndEncodeHtmlForPdf" should {
+    "show liability start date" in forAll(
```
Comment (liandra-giacomine) [inline test/uk/gov/hmrc/economiccrimelevyregistration/controllers/CheckYourAnswersControllerSpec.scala:453]: can we have a test for when the date is not present? How do we want to handle that?

File: test/uk/gov/hmrc/economiccrimelevyregistration/controllers/actions/DataRetrievalActionSpec.scala [hunk 3]
```diff
@@ -68,11 +83,19 @@ class DataRetrievalActionSpec extends SpecBase {
           .thenReturn(Future(updatedRegistration))
         when(mockAppConfig.privateBetaAccessCodes).thenReturn(Seq(privateBetaAccessCode))
 
+        val info = RegistrationAdditionalInfo(
+          registration.internalId,
+          Some(random[Int]),
+          None
+        )
+
+        when(mockRegistrationAdditionalInfoService.get(any())(any())).thenReturn(Future.successful(Some(info)))
```
Comment (liandra-giacomine) [inline test/uk/gov/hmrc/economiccrimelevyregistration/controllers/actions/DataRetrievalActionSpec.scala:92]: can we add a negative case for when mockRegistrationAdditionalInfoService.get doesn't return a successful response

**Review by liandra-giacomine — CHANGES_REQUESTED** [review]
Negative case scenario for tests

**Comment by paddynski-opencast** [comment]
Added tests.

**Review by liandra-giacomine — APPROVED** [review]

**Pull request merged**